### ISSUE TITLE
Dialog and Popover enforceFocus has no affect

### ISSUE THREAD

**Issue by mdebeus** [palantir/blueprint#3445]
#### Environment

- __Package version(s)__: 13.14.x
- __Browser and OS versions__: Windows 10, Chrome  73.0.3683.86 (Official Build) (64-bit)

#### Steps to reproduce

I've created both a popover and a dialog with a variety of controls including check boxes and input controls. I set enforceFocus to true, autoFocus to true. Tabbing between items in the dialog  or the popover does not keep the focus within the dialog/popover. 

For the dialog, you can verify this behavior by doing the following:
1. Go to BluepringJS dialog page - all props are on by default
2. Click ShowDialog
3. Tab through all of the controls

#### Actual behavior
Tabbing past 'Visit the Foundry website' button puts you into the chrome address bar instead of back to the first focus-able item in the dialog.

As an aside, the 'Visit the Foundry website' button has no focus rect (hard to see that it is the focused element)

#### Expected behavior
Tabbing past the `Visit the Foundry website' button sets focus to the first focus-able item in the dialog or popover.

**Comment by adidahiya** [comment]
You're right, it looks like you can exit the dialog at the end by pressing tab repeatedly. This seems like a regression... maybe Chrome changed its behavior here. It's interesting that you can't exit the dialog in reverse with shift-tab.

This is the focus rect I see on the Visit Foundry button:

![image](https://user-images.githubusercontent.com/723999/54974810-a6cc1b80-4f6b-11e9-9bb3-1f702a00d4a7.png)

**Comment by mdebeus** [comment]
The focus rect you show looks like something from FireFox. On Chrome, there is no blue bounding box surrounding the button - instead, the button border is changed to blue - but because the button is blue, this focus rect is not visible.

![BlueprintJsInvisibleFocusRect](https://user-images.githubusercontent.com/6082350/55022595-24b61400-4fb8-11e9-8059-cc1016b9502d.png)

**Comment by mdebeus** [comment]
Accidentally closed the issue - sorry.

**Comment by adidahiya** [comment]
My screenshot is from Chrome 72.0.3626.121 on macOS High Sierra. Haven't tested on a Windows machine lately.

**Comment by mdebeus** [comment]
I am also finding that the Popover.hasBackdrop does not work either. Here is what I have:
<div onKeyDown={handleOnKeyDown}>
{array of Cards, each with an 'edit' button}
</div>

The handleOnKeyDown method catches the up/down/left/right arrows and sets the focus to the next/previous Card as appropriate.

When the Card's edit button is clicked, a Popover (to edit the contents of the card) is displayed with the following properties:
                autoFocus={true}
                canEscapeKeyClose={!busy}
                enforceFocus={true}
                hasBackdrop={true}
                interactionKind={PopoverInteractionKind.CLICK}
                isOpen={isOpen}
                lazy={true}
                position={Position.LEFT_BOTTOM}
                usePortal={true}

When the Popover is open, and I type the up/down/left/right arrows, my handleOnKeyDown method is still called with the keystrokes.
The documentation states: hasBackdrop enables an invisible overlay beneath the popover that captures clicks and prevents interaction with the rest of the document until the popover is closed. 

Am I doing something wrong or is this another bug?
-------------------------------------------------------------------------
I'll add one more bug I found with the popover:
I have modified my GUI so that this is no longer an issue for me now, but when the browser scrollbars were visible and performed the scrolling I had the following problematic scenario:
I had many Cards, such that I had to scroll some of them into view. For those scrolled into view, when I clicked their edit button, the popover would display, but the display of the popover would cause the window to scroll to the top, thus hiding the popover.

**Comment by RetWolf** [comment]
Here are some screenshots from Chrome 73.0.3683.86. There is a focus rect, but it is incredibly hard to see. These screenshots were taken on 300% zoom in Chrome. I am on Windows, so this might be a Windows specific problem.

![DialogAction](https://user-images.githubusercontent.com/28321156/55233524-4a812a00-51ff-11e9-8873-4be4a1a80a23.PNG)
![DialogClose](https://user-images.githubusercontent.com/28321156/55233526-4a812a00-51ff-11e9-9a07-82511bac24b6.

The focus rect bug should probably continue discussion in a separate issue, I just wanted to confirm that the issue wasn't specific to @mdebeus only.
